injector/network_disruption_test.go

```diff
@@ -273,22 +278,65 @@ var _ = Describe("Failure", func() {
 						Namespace: "bar",
 					},
 				}
+
+				podsWatcher := watch.NewFake()
+				servicesWatcher := watch.NewFake()
+
+				k8sClient.PrependWatchReactor("pods", testing.DefaultWatchReactor(podsWatcher, nil))
+				k8sClient.PrependWatchReactor("services", testing.DefaultWatchReactor(servicesWatcher, nil))
+
+				// fake watchers for service handling
+				go func() {
+					// Set up
+					time.Sleep(300 * time.Millisecond)
+					servicesWatcher.Add(fakeService)
+					time.Sleep(300 * time.Millisecond)
+					podsWatcher.Add(fakeEndpoint)
+
+					// Deleting a pod
+					time.Sleep(300 * time.Millisecond)
+					podsWatcher.Delete(fakeEndpoint)
+				}()
+
+			})
+
+			It("should add a filter for every service and pods filtered on", func() {
+				// wait for all the addFilters at the beginning of injection to complete
+				time.Sleep(5 * time.Second)
+				priority := uint32(49153)
+
+				Eventually(func() bool {
+					return tc.AssertCalled(GinkgoT(), "AddFilter", []string{"lo", "eth0", "eth1"}, "1:0", mock.Anything, mock.Anything, "nil", "172.16.0.1/32", 0, 80, "TCP", "1:4")
```

Comment:
Does this work? I've previously seen that `tc.AssertCalled` will directly fail the test, because it's being passed `GinkgoT()`, rather than just returning `false`. But, maybe you're passing `T` better than I was?

Reply:
It works, I didn't encounter this problem